### ISSUE TITLE
SCREENSHARE_XDP does nothing

### ISSUE THREAD

**Issue by checkraisefold** [Nheko-Reborn/nheko#1613]
### Describe the bug

The option SCREENSHARE_XDP in `CMakeLists.txt` added in #1411 has apparently always does nothing.
This is an issue for any potential Windows build in the future supporting gstreamer, since there is no Qt D-Bus API/XDP on Windows, and such these files have to be skipped.

### To Reproduce

1. Use option SCREENSHARE_XDP
2. Does nothing

### Version

Latest master

### Did you use profiles?

- [ ] Profiles used?

**Comment by nico-famedly** [comment]
I guess the idea was to use that as a define in the C++ code, but that seems to have never happened? At least this doesn't require any extra libraries, so it has a somewhat small impact, it would just show a button, that shouldn't exist.

**Comment by checkraisefold** [comment]
> I guess the idea was to use that as a define in the C++ code, but that seems to have never happened? At least this doesn't require any extra libraries, so it has a somewhat small impact, it would just show a button, that shouldn't exist.

Yes, the define should in theory prevent the ScreencastPortal from getting compiled (currently only checks GSTREAMER_AVAILABLE) and also disable stuff that uses ScreencastPortal.
